### DISCUSSION TITLE
attribut __malloc__ does not take arguments during pip install horovod.

### DISCUSSION THREAD

**Discussion by Lana-Deere** [horovod/horovod#3715] [Q&A] [answered]
I'm trying to pip install horovod[tensorflow] on Rocky Linux 9 (RedHat 9 clone) with g++ 11.2.1, python 3.9.13, tensorflow 2.10, cuda 11.5.  Early In the install I get an error,

        #$ cicc --c++17 --gnu_version=110201 --display_error_number
        --orig_src_file_name "CMakeCUDACompilerId.cu" --orig_src_path_name
        "/tmp/pip-install-cbnmwk7i/horovod_8fbd471d53bf4946870a03253832453b/build/temp.linux-x86_64-cpython-39/RelWithDebInfo/CMakeFiles/3.20.2/CompilerIdCUDA/CMakeCUDACompilerId.cu"
        --allow_managed -arch compute_52 -m64 --no-version-ident -ftz=0 -prec_div=1
        -prec_sqrt=1 -fmad=1 --include_file_name "CMakeCUDACompilerId.fatbin.c"
        -tused --gen_module_id_file --module_id_file_name
        "tmp/CMakeCUDACompilerId.module_id" --gen_c_file_name
        "tmp/CMakeCUDACompilerId.cudafe1.c" --stub_file_name
        "tmp/CMakeCUDACompilerId.cudafe1.stub.c" --gen_device_file_name
        "tmp/CMakeCUDACompilerId.cudafe1.gpu" "tmp/CMakeCUDACompilerId.cpp1.ii" -o
        "tmp/CMakeCUDACompilerId.ptx"
      
        /usr/include/stdio.h(183): error: attribute "__malloc__" does not take
        arguments

I haven't been able to figure out why I'm getting it.  My various cuda C++ cuda programs all compile successfully and function correctly, but those are using g++ 7.5.0.  For horovod I'm using g++ 11.2.1 because the documentation says it is necessary to use a newer compiler with Tensorflow 2.10.

Anyone seen this and/or have any suggestions how to fix it?  Thanks.

**Comment by maxhgerlach** [comment, marked as answer]
Hi @Lana-Deere,

that looks just like the error discussed over at NVIDIA here: https://forums.developer.nvidia.com/t/cuda-11-5-samples-throw-multiple-error-attribute-malloc-does-not-take-arguments/192750/8

Have you tried to upgrade your CUDA runtime, for instance to 11.5.2, the most recent patch version of 11.5? A fix should come starting with 11.5.1 or 11.5.2 if that thread is right.

Alternatively, you could try to downgrade to g++ 10 to workaround the bug.

**Reply by Lana-Deere** [reply]
I upgraded to cuda 11.7.1 and was able to build horovod on Rocky 9. without hitting the error.

**Reply by maxhgerlach** [reply]
Great, thanks for confirming that.

**Comment by Lana-Deere** [comment]
I will try upgrading the version of Cuda first and will see if that works.

.. Lana ***@***.***)
